vmm/src/lib.rs

```diff
@@ -1524,7 +1525,12 @@ impl Vmm {
             if drive_config.drive_id == *drive_id {
                 let metadata = metadata(&drive_config.path_on_host)
                     .map_err(|_| DriveError::BlockDeviceUpdateFailed)?;
-                let new_size = metadata.len();
+                let new_size = if metadata.file_type().is_block_device() {
+                    get_blk_dev_size(&drive_config.path_on_host)
+                        .map_err(|_| DriveError::BlockDeviceUpdateFailed)?
+                } else {
+                    metadata.len()
+                };
```

Comment:
I didn't know that but it works! Would it be better to update the file-backed device path as well? Since metadata() wouldn't need open(), I'd like to only update block.rs to use fseek (std::io::Seek in Rust).

Reply:
Addressed on 14ae5459c27c30b3581ea3ed9cfd7afd5169c904

Reply:
This method should work on both files and block devices. With that in mind, do we really need an extra branch in the code to check if the source is a regular file or a block device and an extra function in `fc_util`?

Reply:
`metadata.len()` would be faster...? But yeah, I'm fine inclining open + seek here.